toolchain/sem_ir/formatter.cpp

```diff
@@ -384,15 +384,15 @@ class FormatterImpl {
     FormatParamList(fn.implicit_param_patterns_id, /*is_implicit=*/true);
     FormatParamList(fn.param_patterns_id, /*is_implicit=*/false);
 
-    if (fn.return_slot_id.is_valid()) {
+    if (fn.return_slot_pattern_id.is_valid()) {
       out_ << " -> ";
       auto return_info = ReturnTypeInfo::ForFunction(sem_ir_, fn);
       if (!fn.body_block_ids.empty() && return_info.is_valid() &&
           return_info.has_return_slot()) {
-        FormatName(fn.return_slot_id);
+        FormatName(fn.return_slot_pattern_id);
```

Comment:
Not in this PR, but I think we should revisit how we're formatting the `fn` declarations in SemIR. Currently, we have things like `%n.param_patt` in the parameters list (and now `%return.param_patt` in the return position), but the body refers to `%n` or `%n.loc123` instead, which seems to lose the connection between the two.

Perhaps this would be addressed by the change we discussed recently, of including the callee side of the match IR in the function definition and thus in the `fn` in the SemIR, and instead including the pattern IR in the `function_decl` instruction?